_posts/2018-02-02-artsy-apogee.markdown

```diff
@@ -0,0 +1,65 @@
+---
+layout: epic
+title: "Apogee: Doing More with Less"
+date: 2018-02-02
+categories: [rails]
+author: [ash]
+series: Apogee
+---
+
+> Apogee: the point in the orbit of two objects at which they are furthest apart.
+
+In 2017, the Artsy Auctions Operations team coordinated and ran 190+ sales on our platform. This year, our ambitions are set even higher. But scaling up the number of sales we run will require scaling up our tools and processes, too. This post describes Apogee, a tool I built to help us scale our businesses processes. I never thought I would be so excited to build a spreadsheet plugin, but I honestly had a blast. So let's dive in!
+
+<!-- more -->
+
+Running a sale on Artsy is no small feat. I mean, after all the contract negotiations you might think things get easier, but that's just the beginning of the work. All our auction partners have data in their own CMS systems, and they're all formatted slightly differently. We need to get the information for each lot in a sale into Artsy's CMS, and so a few years ago we built a batch-import app in Rails to do this (closed source, sorry). It works well, but expects data in a certain format.
+
+A lot of work is done by our Ops team to reformat spreadsheets they get from our partners to match the structure our batch import tool expects. All of our partners have different formats, and sales can include hundreds of lots. The reformatting process can take one person over a day, for large sales.
+
+Wouldn't it be cool to build some kind of server to bridge the gulf between Artsy's database and our partner's myriad systems?
+
+No, actually, it wouldn't. It would be a tremendous amount of work, from our side and from theirs. Back to the drawing board.
+
+In January, Ops arranged a meeting between us engineers and an auction partner. [Skinner][] was kind enough to walk us through their export process and provide us with some representative data.
+
+Perfect, now we have a starting point.
+
+We built a quick prototype – less than two days work – to pull out data from Skinner's spreadsheets and format it into the structure that's easiest for our Ops team to work with. The prototype itself was a [Google Sheets Add-on][add-on]. There will be a follow-up post describing the technical evolution of this tool, but the important thing is that we engineers went to where our Ops team already was. Previous discussions around improving Ops' workflow were centred around building entirely new systems, instead of building tools that fit into our existing, functional workflows.
+
+The prototype was used with two large sales Skinner and Artsy were running together. Parsing out _just_ the dimensions of the lots, for _just_ one of the sales, saved an hour of Ops' time. Clearly, there was promise in this tool.
```

Comment:
Three sales, I think

Reply:
I got Leslie to proofread the post, and apparently two of the sales got merged.